src/bitfield.rs

```diff
@@ -235,6 +236,32 @@ impl<N: Unsigned + Clone> Bitfield<Variable<N>> {
         }
         result
     }
+
+    /// Returns `true` if `self` is a subset of `other` and `false` otherwise.
+    pub fn is_subset(&self, other: &Self) -> bool {
+        match self.len().cmp(&other.len()) {
+            Ordering::Equal | Ordering::Less => {
+                let intersection = self.intersection(other);
+                intersection == *self
+            }
+            // We handle the case where self.len() is greater than other.len() differently to account
+            // for BitLists where the higher bytes are all 0.
+            // Consider the case where
+            // `a = 0b1011 0b0010 0b0000`
+            // `b = 0b1011 0b0010`
+            // Here, `a` is a subset of `b` but if we handle it similarly to the `Ordering::Equal` and `Ordering::Less` case
+            // we get `a.intersection(b) = 0b1011 0b0010 ≠ a` and `a.is_subset(&b)` would return `false`.
+            Ordering::Greater => {
+                let mut result =
+                    Self::with_capacity(self.len()).expect("max len always less than N");
+                for i in 0..other.bytes.len() {
+                    result.bytes[i] = other.bytes.get(i).copied().unwrap_or(0);
+                }
+                let intersection = self.intersection(&result);
+                intersection == *self
```

Comment:
can this maybe be written as `self.difference(other).is_zero()`?

Reply:
Yep, this is much cleaner and would work in all cases I think. 
Thank you!